server/pairing/connection.go

```diff
@@ -45,21 +51,61 @@ func NewConnectionParams(netIPs []net.IP, port int, publicKey *ecdsa.PublicKey,
 //   - string type identifier
 //   - version
 //   - net.IP
-//   - version 1: a single net.IP
-//   - version 2: array of IPs in next form:
+//   - array of IPs in next form:
 //     | 1 byte | 4*N bytes | 1 byte | 16*N bytes |
 //     |   N    | N * IPv4  |    M   |  M * IPv6  |
 //   - port
 //   - ecdsa CompressedPublicKey
 //   - AES encryption key
+//   - string InstallationID of the sending device
+//   - string KeyUID of the sending device
+// NOTE:
+// - append(accrete) parameters instead of changing(breaking) existing parameters. Appending should **never** break, modifying existing parameters will break. Watch this before making changes: https://www.youtube.com/watch?v=oyLBGkS5ICk
+// - never strictly check version, unless you really want to break
+
+// This flag is used to keep compatibility with 2.29. It will output a 5 parameters connection string with version 3.
+var keep229Compatibility bool = true
```

Comment:
oh, that is not very good 😮

I understand it's only used once and should be fine. But in general, modifying global variable in tests is not a good idea. it might lead to flaky tests.

It's better to move this flag inside the `ConnectionParams`.
And then modify it in test like this:
```go
func (s *ConnectionParamsSuite) TestConnectionParams_ToString() {
	cp, err := s.server.MakeConnectionParams()
	s.Require().NoError(err)
        
        cp.keep229Compatibility = false

	cps := cp.ToString()
	s.Require().Equal(connectionString, cps)
}
```

Reply:
Ok no worries then. It's just a general heads up 😄 
In this case looks fine anyway